### DISCUSSION TITLE
curl on Windows(Schannel) fails in AcquireCredentialsHandle when running under NT AUTHORITY\Local Service

### DISCUSSION THREAD

**Discussion by jayvpp** [curl/curl#8729] [Q&A]
I'm not sure if this is a bug or it is the expected behavior. I need some help here.

I have a service running under NT AUTHORITY\Local Service that uses curl.
Curl is returning error (SEC_NO_CREDENTIALS) in this line: s_pSecFn->AcquireCredentialsHandle(NULL, (TCHAR *)UNISP_NAME,...) 
link below
https://github.com/curl/curl/blob/f7f26077bc563375becdb2adbcd49eb9f28590f9/lib/vtls/schannel.c#L724

When I run the same piece of code under a local user account (using a test app), everything works fine. 

Why does running the code in a service (in session 0) in the context of NT AUTHORITY\Local Service causes this issue?
Is there a way to fix this?

Regards

**Comment by gvanem** [comment]
[MSDN](https://docs.microsoft.com/en-us/windows/win32/secauthn/acquirecredentialshandle--credssp#parameters) says:
*A null string indicates that the process requires a handle to the credentials of the user under whose security context it is executing.*

So in your case a NULL-string is not sufficient.

**Reply by jayvpp** [reply]
yes, but I do not own that code(it is curl code) and don't see any flag available to be able to pass another parameter.

Is there a way (e.g using Impersonation) or some other mechanism available so my process/service can get access to the credentials of the user under whose security context it is executing.

Regards

Thanks

**Comment by jay** [comment]
According to the documentation when using Schannel that parameter is always supposed to be [set to NULL](https://docs.microsoft.com/en-us/windows/win32/secauthn/acquirecredentialshandle--schannel#parameters) so I'm not sure what you're supposed to do in this case. Probably your system process needs to acquire credentials to some other user, or run a child process as some other user. That is beyond the scope of what we can help you with here.

**Reply by jayvpp** [reply]
Ok, Thanks
